### PR TITLE
#9 Task. TapeGhad

### PR DESCRIPTION
**Pull request by TapeGhad**
https://travis-ci.com/github/TapeGhad/js-assignments/builds/174479217

### CODE REVIEW

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -65,7 +67,13 @@ function getPowerFunction(exponent) {
  *   getPolynom()      => null
  */
 function getPolynom() {
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js:69]: https://javascript.info/coding-style

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -84,7 +92,8 @@ function getPolynom() {
  *   memoizer() => the same random number  (next run, returns the previous cached result)
  */
 function memoize(func) {
-    throw new Error('Not implemented');
+    let value=func()
+    return ()=>{return value}
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js]: use simplified version of arrow function () => value

File: task/09-functions-n-closures-tasks.js [hunk 3]
```diff
@@ -104,7 +113,17 @@ function memoize(func) {
  * retryer() => 2
  */
 function retry(func, attempts) {
-    throw new Error('Not implemented');
+    return function TryFunc(){
+        while (attempts>0){
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js]: You have 2 exit conditions. you need only 1 in the loop definition. Also use for-loop

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/09-functions-n-closures-tasks.js [hunk 4]
```diff
@@ -104,7 +115,17 @@ function memoize(func) {
  * retryer() => 2
  */
 function retry(func, attempts) {
-    throw new Error('Not implemented');
+    return function TryFunc(){
+        for (; attempts > 0;){
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js]: just use a normal for(let i =0; exitcondition; i++) loop

File: task/09-functions-n-closures-tasks.js [hunk 5]
```diff
@@ -104,7 +115,17 @@ function memoize(func) {
...
+            try {
+                return func();
+            }
+            catch (e){
+                attempts-=1;
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js]: Do not modify funciton arguments if not required. 99% of the time you don't need to do it.

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by TapeGhad** [comment]
Fixed

File: task/09-functions-n-closures-tasks.js (hunk 5 above)
Comment (TapeGhad) [reply task/09-functions-n-closures-tasks.js]: Fixed

File: task/09-functions-n-closures-tasks.js (hunk 4 above)
Comment (TapeGhad) [reply task/09-functions-n-closures-tasks.js]: Fixed

Comment (AlexMokin) [reply task/09-functions-n-closures-tasks.js]: not fixed. Increment should also be in the loop defenition. A general note 99% of the time you should keep increment in the loop definition. This way it's way easier to read and most of the time it's unnecessary to move it in different place.

**Comment by AlexMokin** [comment]
Added comment to the retry function.

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by TapeGhad** [comment]
I can't find comment again. In FILES CHANGED there is no any info too
image

**Comment by TapeGhad** [comment]
Retry function is changed a little bit

**Comment by AlexMokin** [comment]
фссузеув

**Pull request closed without merging**